ui/app/registration/controllers/patientCommonController.js

```diff
@@ -39,6 +45,49 @@ angular.module('bahmni.registration')
                 }
             });
 
+            $scope.checkDuplicatePatients = function () {
+                var patientGivenName = $scope.patient.givenName || '';
+                var patientLastName = $scope.patient.familyName || '';
+                var gender = $scope.patient.gender || '';
+                var birthDate = $scope.patient.birthdate || '';
+                var contactNumber = $scope.patient.PERSON_ATTRIBUTE_TYPE_PHONE_NUMBER || '';
+                if (birthDate != '') {
+                    birthDate = new Date(birthDate);
+                    birthDate = convertDate(birthDate);
+                }
+                var queryParams = patientGivenName + ' ' + patientLastName;
+
+                if (queryParams.length > 1) {
+                    patientService.searchDuplicatePatients(queryParams, gender, birthDate, contactNumber).then(function (response) {
+                        
+                        _.map(response.pageOfResults, function (result) {
+                            result.addressFieldValue = JSON.parse(result.addressFieldValue);
+                            result.extraIdentifiers = JSON.parse(result.extraIdentifiers);
+                            result.customAttribute = JSON.parse(result.customAttribute);
+                        });
+
+                        $rootScope.patients = response.pageOfResults;
+                        $rootScope.countNo = $rootScope.patients.length;
+                        $rootScope.visibleData = true;
+
+                        if ($rootScope.countNo == 0) {
+                            $rootScope.visibleData = false;
+                        }
+                    });
+                } else {
+                    $rootScope.visibleData = false;
```

Comment:
@IipperMangesh Can we use a more explicit variable name here (i.e. which data is being set visible)?

Reply:
Yes, changed its name to 'duplicatePatientVisibleData'.